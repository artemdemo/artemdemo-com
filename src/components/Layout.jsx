import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'gatsby';
import Helmet from 'react-helmet';
import Container from './Container';

class Layout extends React.Component {
    renderHeader() {
        const { location, metaTitle } = this.props;
        const rootPath = `${__PATH_PREFIX__}/`;

        if (location.pathname === rootPath) {
            return (
                <h1>{metaTitle}</h1>
            )
        }

        return (
            <h3>
                <Link to={'/'}>
                    {metaTitle}
                </Link>
            </h3>
        );
    }

    render() {
        const { title, description } = this.props;
        return (
            <Container>
                <Helmet
                    htmlAttributes={{lang: 'en'}}
                    meta={[{name: 'description', content: description}]}
                    title={title}
                />
                {this.renderHeader()}
                {this.props.children}
            </Container>
        )
    }
}

Layout.propTypes = {
    location: PropTypes.shape({
        pathname: PropTypes.string,
    }),
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    metaTitle: PropTypes.string.isRequired,
};

Layout.defaultProps = {
    location: {},
};

export default Layout
