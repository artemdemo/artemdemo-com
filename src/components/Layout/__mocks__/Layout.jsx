import React from 'react';

const Layout = props => (
    <div data-mock='Layout'>
        {props.children}
    </div>
);

export default Layout;