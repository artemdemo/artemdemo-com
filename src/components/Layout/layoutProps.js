import PropTypes from 'prop-types';

export const locationProp = PropTypes.shape({
  pathname: PropTypes.string,
});
