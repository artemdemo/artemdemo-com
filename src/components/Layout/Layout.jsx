import React from 'react';
import PropTypes from 'prop-types';
import Helmet from 'react-helmet';
import Container from '../Container/Container';
import Header from '../Header/Header';
import { locationProp } from './layoutProps';
import pkg from '../../../package.json';

class Layout extends React.Component {
    render() {
        const { title, description, location } = this.props;
        return (
            <React.Fragment>
                <Helmet
                    htmlAttributes={{lang: 'en'}}
                    meta={[
                        {name: 'description', content: description},
                        {name: 'app-version', content: pkg.version},
                    ]}
                    title={title}
                />
                <Header location={location} />
                <Container>
                    {this.props.children}
                </Container>
            </React.Fragment>
        )
    }
}

Layout.propTypes = {
    location: locationProp,
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    metaTitle: PropTypes.string.isRequired,
};

Layout.defaultProps = {
    location: {},
};

export default Layout
