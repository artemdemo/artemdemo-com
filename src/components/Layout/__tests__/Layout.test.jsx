import React from 'react';
import renderer from 'react-test-renderer';
import Layout from '../Layout';

jest.mock('react-helmet');
jest.mock('../../Icon/Icon');
jest.mock('../../Header/Header');

describe('Layout', () => {
    it('should render empty', () => {
        const tree = renderer.create(
            <Layout
                metaTitle='Meta title'
                title='Regular title'
                description='Site description'
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render with children', () => {
        const tree = renderer.create(
            <Layout
                metaTitle='Meta title'
                title='Regular title'
                description='Site description'
            >
                <div>Some child</div>
            </Layout>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});