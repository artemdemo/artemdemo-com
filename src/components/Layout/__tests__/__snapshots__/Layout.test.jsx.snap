// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render empty 1`] = `
Array [
  <div
    data-mock="Helmet"
  >
    {
  "htmlAttributes": {
    "lang": "en"
  },
  "meta": [
    {
      "name": "description",
      "content": "Site description"
    },
    {
      "name": "app-version",
      "content": "0.0.0-mock"
    }
  ],
  "title": "Regular title"
}
  </div>,
  <div
    data-mock="Header"
  >
    {
  "title": "Regular title",
  "location": {}
}
  </div>,
  <div
    data-mock="Container"
  >
    {}
  </div>,
]
`;

exports[`Layout should render with children 1`] = `
Array [
  <div
    data-mock="Helmet"
  >
    {
  "htmlAttributes": {
    "lang": "en"
  },
  "meta": [
    {
      "name": "description",
      "content": "Site description"
    },
    {
      "name": "app-version",
      "content": "0.0.0-mock"
    }
  ],
  "title": "Regular title"
}
  </div>,
  <div
    data-mock="Header"
  >
    {
  "title": "Regular title",
  "location": {}
}
  </div>,
  <div
    data-mock="Container"
  >
    {}
    <div>
      Some child
    </div>
  </div>,
]
`;
